x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/PolicyStepsRegistry.java

```diff
@@ -61,12 +74,36 @@ public PolicyStepsRegistry() {
 
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    public void update(IndexLifecycleMetadata meta, Client client, LongSupplier nowSupplier) {
+    public void update(ClusterState clusterState, IndexLifecycleMetadata meta, Client client, LongSupplier nowSupplier) {
```

Comment:
should we break up this update method into its underlying parts of the MetaData that it is concerned with? Also fine with leaving it as is, just feels big and difficult to juggle which we're updating

Reply:
I think that's a good idea, but not yet. The next parts of this refactor are going to be touching this method quite a bit, so I don't think it makes sense to split it up quite yet